src/snapred/backend/recipe/algorithm/WriteWorkspaceMetadata.py

```diff
@@ -0,0 +1,76 @@
+import json
+from typing import Dict
+
+from mantid.api import (
+    AlgorithmFactory,
+    MatrixWorkspaceProperty,
+    PropertyMode,
+    PythonAlgorithm,
+)
+from mantid.kernel import Direction
+
+from snapred.backend.dao.WorkspaceMetadata import UNSET, WorkspaceMetadata
+from snapred.backend.recipe.algorithm.MantidSnapper import MantidSnapper
+from snapred.backend.recipe.algorithm.ReadWorkspaceMetadata import ReadWorkspaceMetadata
+
+
+class WriteWorkspaceMetadata(PythonAlgorithm):
+    def category(self):
+        return "SNAPRed Internal"
+
+    def PyInit(self):
+        # declare properties
+        self.declareProperty(
+            MatrixWorkspaceProperty("Workspace", "", Direction.Input, PropertyMode.Mandatory),
+            doc="Workspace to contain the logs",
+        )
+        self.declareProperty("WorkspaceMetadata", defaultValue="", direction=Direction.Input)
+        # self.declareProperty("MetadataLogName", defaultValue=[], direction=Direction.Input) # string array property
```

Comment:
do you need these for later at some point?

Reply:
I can get rid of them for now, but Kort had suggested a way to write logs without having to use an object, and that's what those would eventually be used for.  I plan to implement those as we work on the grocery service methods